package com.owo233.fuckmarketads.apps

import com.owo233.fuckmarketads.hooks.market.HideSecurityView
import com.owo233.fuckmarketads.hooks.market.MiscApply
import com.owo233.fuckmarketads.hooks.market.RemoveAds
import com.owo233.fuckmarketads.hooks.market.TabFilter
import com.owo233.fuckmarketads.init.AppPackage
import com.owo233.fuckmarketads.init.AppRegister
import com.owo233.fuckmarketads.HookSettings
import io.github.libxposed.api.XposedModuleInterface

object Market : AppRegister() {

    override val packageName: String
        get() = AppPackage.MARKET

    override fun onPackageReady(param: XposedModuleInterface.PackageReadyParam) {
        HookSettings.init(HookEnv.base)
        autoInitHooks(
            param,
            HideSecurityView,
            TaFilter,
            RemoveAds,
            MiscApply
        )
    }
}